File: stock_price.py
import sys
import argparse

def get_last_close_price(ticker):
    """
    Get the last closing price for a given stock ticker.
    Returns the price as a float rounded to 2 decimal places, or None if not found.
    """
    try:
        import yfinance as yf
        stock = yf.Ticker(ticker.upper())
        # Get historical data for the last 5 days to ensure we have data
        hist = stock.history(period="5d")
        if hist.empty:
            return None
        # Get the last close price
        last_close = hist['Close'].iloc[-1]
        return round(last_close, 2)
    except ImportError:
        # yfinance is not installed, use mock data
        mock_prices = {
            'AAPL': 175.43,
            'MSFT': 342.12,
            'GOOGL': 142.78,
            'AMZN': 185.67,
            'TSLA': 248.91,
            'META': 498.32,
            'NVDA': 925.50,
            'JPM': 198.45,
            'V': 278.11,
            'WMT': 65.89
        }
        return mock_prices.get(ticker.upper(), None)
    except Exception as e:
        # Other errors (like network issues) - fall back to mock
        mock_prices = {
            'AAPL': 175.43,
            'MSFT': 342.12,
            'GOOGL': 142.78,
            'AMZN': 185.67,
            'TSLA': 248.91,
            'META': 498.32,
            'NVDA': 925.50,
            'JPM': 198.45,
            'V': 278.11,
            'WMT': 65.89
        }
        return mock_prices.get(ticker.upper(), None)

def main():
    parser = argparse.ArgumentParser(description='Get the last close price for a US stock symbol.')
    parser.add_argument('ticker', help='Stock ticker symbol (e.g., AAPL, MSFT)')
    args = parser.parse_args()
    
    price = get_last_close_price(args.ticker)
    if price is None:
        print(f"Error: Could not retrieve price for symbol '{args.ticker}'", file=sys.stderr)
        sys.exit(1)
    
    print(f"{args.ticker.upper()}: ${price}")

if __name__ == '__main__':
    main()

File: src/StockPrice.java
import java.util.HashMap;
import java.util.Map;
import java.util.Optional;

public class StockPrice {
    private static final Map<String, Double> MOCK_PRICES = new HashMap<>();

    static {
        MOCK_PRICES.put("AAPL",  175.43);
        MOCK_PRICES.put("MSFT",  342.12);
        MOCK_PRICES.put("GOOGL", 142.78);
        MOCK_PRICES.put("AMZN",  185.67);
        MOCK_PRICES.put("TSLA",  248.91);
        MOCK_PRICES.put("META",  498.32);
        MOCK_PRICES.put("NVDA",  925.50);
        MOCK_PRICES.put("JPM",   198.45);
        MOCK_PRICES.put("V",     278.11);
        MOCK_PRICES.put("WMT",    65.89);
    }

    public static Optional<Double> getLastClosePrice(String ticker) {
        if (ticker == null || ticker.isEmpty()) {
            return Optional.empty();
        }
        return Optional.ofNullable(MOCK_PRICES.get(ticker.toUpperCase()));
    }

    public static void main(String[] args) {
        if (args.length != 1) {
            System.err.println("Usage: stock_price <ticker>");
            System.exit(1);
        }

        String ticker = args[0];
        Optional<Double> price = getLastClosePrice(ticker);
        if (!price.isPresent()) {
            System.err.printf("Error: Could not retrieve price for symbol '%s'%n", ticker);
            System.exit(1);
        }

        System.out.printf("%s: $%.2f%n", ticker.toUpperCase(), price.get());
    }
}
